components/esp_ringbuf/include/freertos/ringbuf.h

```diff
@@ -496,6 +496,16 @@ void vRingbufferGetInfo(RingbufHandle_t xRingbuffer,
                         UBaseType_t *uxAcquire,
                         UBaseType_t *uxItemsWaiting);
 
+/**
+ * @brief   Unblock any read function that is currently waiting. example: xRingbufferReceiveUpTo()
+ * 
+ * All read functions take a xTicksToWait argument, which can be set up to
+ * to infinity. This function will unblock any threads currently waiting.
+ *
+ * @param[in]   xRingbuffer     The ring buffer who's rx will be unblocked.
+ */
+void vRingbufferUnblockRx(RingbufHandle_t xRingbuffer);
```

Comment:
> If usb_serial_jtag.c starts using something other than a ringbuffer internally (e.g. a StreamBuffer), we would have to add similar API for unblocking there; and given that StreamBuffer is upstream FreeRTOS code we would most likely want to avoid that.

^ From Ivan. 

Can you re-evaluate your position on this? `vRingbufferUnblockRx` seems like the ideal solution.

If we ever switched to `StreamBuffer` we could come up with alternative solutions. For example:
- A) a counting semaphore + StreamBuffer. 'give' the semaphore on new data + then call StreamBuffer with a 0 timeout.
- B) Use a long timeout with StreamBuffer, perhaps 10 seconds, and poll an exit condition
- C) Make a Pull Request on FreeRTOS to add the necessary unblock feature to StreamBuffer

Reply:
Alternatively, ^^ solution (A), could be used to mitigate the need for `vRingbufferUnblockRx` in this PR as well. 

I think the solution would look like this:
- **normal exclusive mutex.**  `usb_serial_jtag_read_bytes` will grab this mutex first. _Not entirely necessary._ This makes sure the counting semaphore only has a single thread using it at a time. **i.e.** so the first thread to call `read()` will always be satisfied first.
- **counting semaphore.** `usb_serial_jtag_isr_handler_default` increments this counting semaphore _once_ each time it appends to the ringbuffer, `usb_serial_jtag_read_bytes` will 'take' from the semaphore, in order to wait until new bytes are ready (We could also use `event_groups` instead).
- **ring buffer**, `usb_serial_jtag_read_bytes` calls the ringbuffer with a zero timeout, and will wait on the counting semaphore again until there is enough data.

Of course, I would consider the `vRingbufferUnblockRx` solution better.